eng/pipelines/docindex.yml

```diff
@@ -18,12 +18,15 @@ jobs:
 
       - pwsh: |
           Invoke-WebRequest -Uri "https://github.com/dotnet/docfx/releases/download/v2.43.2/docfx.zip" `
-          -OutFile "docfx.zip" | Wait-Process; Expand-Archive -Path "docfx.zip" -DestinationPath "./docfx/"
+          -OutFile "docfx.zip" | Wait-Process; Expand-Archive -Path "docfx.zip" -DestinationPath ./docfx
+          echo "##vso[task.setvariable variable=docfxPath;isOutput=true]$(Build.SourcesDirectory)/docfx/docfx.exe"
         workingDirectory: $(Build.SourcesDirectory)
         displayName: Download and Extract DocFX
 
       - pwsh: |
```

Comment:
Since this is only a ps script call lets turn this into the PowershellTask approach so we get better command line diagnostics.

Reply:
```
- task: PowerShell@2
  displayName: 'Generate Doc Index'
  inputs:
    pwsh: true
    filePath: $(Build.SourcesDirectory)/eng/docgeneration/Generate-DocIndex.ps1 
    arguments: >
      -Docfx "$(docfxPath)"
      -RepoRoot $(Build.SourcesDirectory)
      -DocGenDir "$(Build.SourcesDirectory)/eng/docgeneration"
      -verbose
```